packages/loader/container-loader/src/container.ts

```diff
@@ -752,7 +752,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         // runtime matches pending ops to successful ones by clientId and client seq num, so we need to close the
         // container at the same time we get pending state, otherwise this container could reconnect and resubmit with
         // a new clientId and a future container using stale pending state without the new clientId would resubmit them
-        this._deltaManager.close();
+        this.forceReadonly(true);
```

Comment:
The reason for this is to work around a race condition while maintaining the spirit of the original change.

The reason `this._deltaManager.close();` won't work anymore is that on close, the delta manager will close the container. When this happens, the container is also closed (as configured). In turn, the context is disposed along with the state manager (who is the subject of this PR). This call chain will clear all pending state and make this function useless. 

This has been uncovered by `packages/test/test-end-to-end-tests/src/test/pendingOps.spec.ts`

Reply:
Makes sense. Should this call be removed then? I.e. there seems like no benefit from having this call, right?

Reply:
That's a great question. I guess the point here is to stop accepting ops while we're collecting the pending states, right? Considering there's no async operation 'during' this activity it's safe to assume that we're not going to anyway - correct me if I'm wrong. So I guess this is safe to remove completely?

Reply:
I'd remove it (as otherwise next person will be asking same questions - what purpose it solves?).
readonly is not going to stop ops from coming. If we want to disable that, we need to pause inboud queue.

Reply:
Will rework this and try to pause the queue then..